nvi-commons/src/main/java/no/sikt/nva/nvi/common/service/model/Candidate.java

```diff
@@ -135,12 +135,10 @@ public static Candidate fetch(FetchCandidateRequest request, CandidateRepository
         return new Candidate(repository, candidateDao, approvalDaoList, noteDaoList, periodStatus);
     }
 
-    public static Optional<Candidate> upsert(UpsertCandidateRequest request, CandidateRepository repository,
-                                             PeriodRepository periodRepository) {
+    public static void upsert(UpsertCandidateRequest request, CandidateRepository repository) {
         var optionalCandidate = repository.findByPublicationId(request.publicationId());
-        return optionalCandidate.map(
-                candidateDao -> handleExistingCandidate(request, repository, periodRepository, candidateDao))
-                   .orElseGet(() -> Optional.of(createCandidate(request, repository, periodRepository)));
+        optionalCandidate.ifPresentOrElse(candidateDao -> updateExistingCandidate(request, repository, candidateDao),
+                                          () -> createCandidate(request, repository));
     }
 
     public static Optional<Candidate> updateNonCandidate(UpdateNonCandidateRequest request,
```

Comment:
Should this return `void` too? It doesn't seem like the return value is used in production code here either.

Reply:
Probably, future PR?